src/sick_mrs1000_communication.cpp

```diff
@@ -112,23 +128,29 @@ int SickMrs1000Communication::loopOnce()
       {
         ROS_DEBUG_STREAM("Publish cloud with " << cloud.height * cloud.width
           << " points in the frame \"" << cloud.header.frame_id << "\".");
-        diagnosed_cloud_publisher_.publish(cloud);
+        cloud_pub_.publish(cloud);
+
       }
 
+
       /*
        * scan.header.frame_id == "" means the laser scan was of a layer
        * different than layer 0, so don't publish it (because the points
        * don't lie in a plane)
        */
       if(scan.header.frame_id != "")
       {
-        diagnosticPub_->publish(scan);
+        pub_.publish(scan);
       }
-
     }
     buffer_pos = dend + 1;
   }
 
+  output_cloud_diagnostic_->tick();
```

Comment:
These ticks should be right beside the publish calls. Because the publish calls are wrapped in if cases, `tick()` could be called without `publish()` actually being called.

Reply:
Good point moved under publish